repositories/strategies/routingStrategy.js

```diff
@@ -0,0 +1,322 @@
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const insertRoutingRuleSet = async (
+  trx,
+  { questionPageId, routingDestinationId }
+) =>
+  trx("Routing_RuleSets")
+    .insert(
+      toDb({
+        questionPageId: parseInt(questionPageId),
+        routingDestinationId: parseInt(routingDestinationId)
+      })
+    )
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const checkAnswerBelongsToPage = async (trx, answerId, questionPageId) => {
+  if (isNil(answerId)) {
+    return;
+  }
+
+  const answer = await findAnswerByIdAndQuestionPageId(
+    trx,
+    answerId,
+    questionPageId
+  );
+  if (!answer) {
+    throw new Error(
+      `Answer ${answerId} does not belong to ${questionPageId}. Choose a different answer.`
+    );
+  }
+};
+
+const getAnswerOrFirstAnswerOnPage = async (trx, answerId, questionPageId) => {
+  if (!isNil(answerId)) {
+    return answerId;
+  }
+  const firstAnswer = await getFirstAnswer(trx, questionPageId);
+  return get(firstAnswer, "id", null);
+};
+
+const updateRoutingConditionStrategy = async (
+  trx,
+  { id: routingConditionId, questionPageId, answerId }
+) => {
+  await checkAnswerBelongsToPage(trx, answerId, questionPageId);
+  const routingConditionAnswer = await getAnswerOrFirstAnswerOnPage(
+    trx,
+    answerId,
+    questionPageId
+  );
+  await deleteRoutingConditionValues(trx, routingConditionId);
+  return updateRoutingCondition(
+    trx,
+    routingConditionId,
+    questionPageId,
+    routingConditionAnswer
+  );
+};
+
+const toggleConditionValueStrategy = async (
+  trx,
+  { conditionId, optionId, checked }
+) => {
+  const table = trx("Routing_ConditionValues");
+  const args = toDb({ conditionId, optionId });
+  const existing = await table.where(args);
+
+  if (checked && !isEmpty(existing)) {
+    throw new Error(
+      `Option ${optionId} already checked for routing condition ${conditionId}`
+    );
+  }
+
+  const query = checked ? table.insert(args) : table.where(args).del();
+  return query
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+};
+
+async function getAvailableRoutingDestinations(trx, pageId) {
+  const page = await getPageById(trx, pageId);
+  const section = await getSectionById(trx, page.sectionId);
+  const questionPages = await getPageDestinations(trx, page);
+  const sections = await getSectionDestinations(trx, section);
+
+  return {
+    questionPages,
+    sections
+  };
+}
+
+const checkRoutingDestinations = async (
+  availableRoutingDestinations,
+  destination
+) => {
+  const { logicalDestinations } = availableRoutingDestinations;
+  const { logicalDestination, absoluteDestination } = destination;
+
+  if (!isNil(logicalDestination)) {
+    if (
+      !find(logicalDestinations, {
+        logicalDestination: logicalDestination.destinationType
+      })
+    ) {
+      throw new Error(
+        `Unable to route from this question ${logicalDestination.destinationType}`
+      );
+    }
```

Comment:
Could you do this in one line using `if(includes(logicalDestinations, logicalDestination.destinationType))`